packages/react-native/Libraries/Text/TextInput/Multiline/RCTWrappedTextView.m

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#if TARGET_OS_OSX // [macOS
+
+#import <React/RCTWrappedTextView.h>
+
+#import <React/RCTUITextView.h>
+#import <React/RCTTextAttributes.h>
+
+@implementation RCTWrappedTextView {
+  RCTUITextView *_forwardingTextView;
+  RCTUIScrollView *_scrollView;
+  RCTClipView *_clipView;
+}
+
+- (instancetype)initWithFrame:(CGRect)frame
+{
+  if (self = [super initWithFrame:frame]) {
+    self.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
+
+    self.hideVerticalScrollIndicator = NO;
+
+    _scrollView = [[RCTUIScrollView alloc] initWithFrame:self.bounds];
+    _scrollView.backgroundColor = [RCTUIColor clearColor];
+    _scrollView.drawsBackground = NO;
+    _scrollView.borderType = NSNoBorder;
+    _scrollView.hasHorizontalRuler = NO;
+    _scrollView.hasVerticalRuler = NO;
+    _scrollView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
+    [_scrollView setHasVerticalScroller:YES];
+    [_scrollView setHasHorizontalScroller:NO];
+    
+    _clipView = [[RCTClipView alloc] initWithFrame:_scrollView.bounds];
+    [_scrollView setContentView:_clipView];
+    
+    _forwardingTextView = [[RCTUITextView alloc] initWithFrame:_scrollView.bounds];
+    _forwardingTextView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
+    _forwardingTextView.delegate = self;
+    
+    _forwardingTextView.verticallyResizable = YES;
+    _forwardingTextView.horizontallyResizable = YES;
+    _forwardingTextView.textContainer.containerSize = NSMakeSize(FLT_MAX, FLT_MAX);
+    _forwardingTextView.textContainer.widthTracksTextView = YES;
+    _forwardingTextView.textInputDelegate = self;
+    
+    _scrollView.documentView = _forwardingTextView;
+    _scrollView.contentView.postsBoundsChangedNotifications = YES;
+    
+    // Enable the focus ring by default
+    _scrollView.enableFocusRing = YES;
+    [self addSubview:_scrollView];
+    
+    // a register for those notifications on the content view.
+  #if !TARGET_OS_OSX // [macOS]

```

Comment:
Remove mobile code since this file is macOS only. @lenaic should we nest these files in a macos specific folder?